Ghidra/Features/Decompiler/src/decompile/cpp/pcodecompile.cc

```diff
@@ -728,8 +738,12 @@ ExprTree *PcodeCompile::createBitRange(SpecificSymbol *sym,uint4 bitoffset,uint4
   }
 
   uintb mask = (uintb)2;
-  mask = ((mask<<(numbits-1))-1);
-  
+  if (numbits - 1 < sizeof(mask) * 8)
```

Comment:
same
```C
uintb mask = 0;
if (numbits -1 < sizeof(mask) * 8)
    mask = (uintb)2 << (numbits - 1);
--mask;
```
could also get moved to this if block
```C
  if (maskneeded)                                                                                    
    appendOp(CPUI_INT_AND,res,mask,finalsize);  
 ```

Reply:
[23bd3d5](https://github.com/NationalSecurityAgency/ghidra/pull/3967/commits/23bd3d5672e81c6288f0585b7271a4f2366494eb)